src/pages/resultsView/ResultsViewPageHelpers.ts

```diff
@@ -0,0 +1,141 @@
+        if (!resultsViewPageStore.selectedMolecularProfileIds || !_.isEqual(resultsViewPageStore.selectedMolecularProfileIds.slice(), profiles)) {
+            resultsViewPageStore.selectedMolecularProfileIds = profiles;
+        }
+
+        if (!_.isEqual(query.RPPA_SCORE_THRESHOLD, resultsViewPageStore.rppaScoreThreshold)) {
+            resultsViewPageStore.rppaScoreThreshold = parseFloat(query.RPPA_SCORE_THRESHOLD);
+        }
+
+        if (!_.isEqual(query.Z_SCORE_THRESHOLD, resultsViewPageStore.zScoreThreshold)) {
+            resultsViewPageStore.zScoreThreshold = parseFloat(query.Z_SCORE_THRESHOLD);
+        }
+
+        if (query.geneset_list) {
+            // we have to trim because for some reason we get a single space from submission
+            const parsedGeneSetList = query.geneset_list.trim().length ? (query.geneset_list.trim().split(/\s+/)) : [];
+            if (!_.isEqual(parsedGeneSetList, resultsViewPageStore.genesetIds)) {
+                resultsViewPageStore.genesetIds = parsedGeneSetList;
+            }
+        }
+
+        // cohortIdsList will contain virtual study ids (physicalstudies will contain the phsyical studies which comprise the virtual studies)
+        // although resultsViewStore doesn
+        if (!resultsViewPageStore.cohortIdsList || !_.isEqual(resultsViewPageStore.cohortIdsList.slice(), cohortIdsList)) {
+            resultsViewPageStore.cohortIdsList = cohortIdsList;
+        }
+
+        if (resultsViewPageStore.oqlQuery !== oql) {
+            resultsViewPageStore.oqlQuery = oql;
+        }
+
+}
+
+export function getVirtualStudies(cancerStudyIds:string[]):Promise<VirtualStudy[]>{
+
+    const prom = new Promise<VirtualStudy[]>((resolve, reject)=>{
+        client.getAllStudiesUsingGET({projection:"SUMMARY"}).then((allStudies)=>{
+            //console.log(cancerStudyIds);
+            //console.log(allStudies);
+            const virtualStudyIds = _.differenceWith(cancerStudyIds, allStudies,(id:string, study:CancerStudy)=>id==study.studyId);
+
+            if (virtualStudyIds.length > 0) {
+                Promise.all(virtualStudyIds.map(id =>  sessionServiceClient.getVirtualStudy(id)))
+                    .then((virtualStudies)=>{
+                         resolve(virtualStudies);
+                    })
+            } else {
+                resolve([]);
+            }
+        });
+    });
+    return prom;
+
+}
+
+export function substitutePhysicalStudiesForVirtualStudies(cancerStudyIds:string[], virtualStudies:VirtualStudy[]){
+
+    let physicalStudies:string[] = [];
+
+    //if a study is a virtual study, substitute its physical study ids
+    const virtualStudiesKeyedById = _.keyBy(virtualStudies,(virtualStudy)=>virtualStudy.id);
+    cancerStudyIds.forEach((studyId)=>{
+        if (studyId in virtualStudiesKeyedById) {
+            const virtualStudy = virtualStudiesKeyedById[studyId];
+            physicalStudies = physicalStudies.concat(virtualStudy.data.origin);
```

Comment:
The origin could be a VS as well

Reply:
Instead of recursively fetching VS info,  we can use all the keys in virtualStudy.data.studies

Reply:
@kalletlak  correct me if I'm wrong

Reply:
yes, origin could be VS.